yt_dlp/extractor/youtube.py

```diff
@@ -3248,6 +3250,20 @@ def _extract_chapters_from_engagement_panel(self, data, duration):
                                           chapter_time, chapter_title, duration)
             for contents in content_list)), [])
 
+    def _extract_heatmap_from_player_overlay(self, data):
+        content_list = traverse_obj(data, (
+            'playerOverlays', 'playerOverlayRenderer', 'decoratedPlayerBarRenderer', 'decoratedPlayerBarRenderer', 'playerBar',
+            'multiMarkersPlayerBarRenderer', 'markersMap', ..., 'value', 'heatmap', 'heatmapRenderer', 'heatMarkers', {list}))
+        scale_mili = functools.partial(float_or_none, scale=1000)
+        generate_end = lambda point: point.get('timeRangeStartMillis') + point.get('markerDurationMillis')
+
+        return next(filter(None, (
+            traverse_obj(contents, (..., 'heatMarkerRenderer', {
+                'start_time': ('timeRangeStartMillis', {scale_mili}),
+                'end_time': ({generate_end}, {scale_mili}),
```

Comment:
```suggestion
        return next(filter(None, (
            traverse_obj(contents, (..., 'heatMarkerRenderer', {
                'start_time': ('timeRangeStartMillis', {scale_mili}),
                'end_time': {lambda x: (x['timeRangeStartMillis'] + x['markerDurationMillis']) / 1000},
```
transformations `{}` are already passed through try_call